### PR TITLE
DLS-11011: [GMP] - Encrypt New SingleCalculationSessionCache Cache

### PR DESCRIPTION
**Pull request by jawadsaleem5747**

### CODE REVIEW

File: app/models/SingleCalculationSessionCache.scala [hunk 1]
```diff
@@ -37,32 +39,41 @@ case class SingleCalculationSessionCache(
...
-        "scenario" -> datedCacheMap.scenario,
-        "revaluationDate" -> datedCacheMap.revaluationDate,
-        "rate" -> datedCacheMap.rate,
-        "leaving" -> datedCacheMap.leaving,
-        "equalise" -> datedCacheMap.equalise,
-        "lastModified" -> datedCacheMap.lastModified
-      )
+    def writes(implicit encryption: Encryption): OWrites[SingleCalculationSessionCache] = new OWrites[SingleCalculationSessionCache] {
+      override def writes(singleCalculationSessionCache: SingleCalculationSessionCache): JsObject = {
+        val encryptedValues: (String, EncryptedValue, EncryptedValue, EncryptedValue, Option[EncryptedValue], Option[EncryptedValue], EncryptedValue, Option[EncryptedValue], Instant) = {
```
Comment (drewmiley) [inline app/models/SingleCalculationSessionCache.scala:60]: Doesn't need {} around ModelEncryption.encryptSingleCalculationSessionCache(singleCalculationSessionCache)

File: conf/application.conf [hunk 2]
```diff
@@ -34,6 +34,9 @@ play.http.errorHandler = "config.MyErrorHandler"
 mongodb {
   uri                 = "mongodb://localhost:27017/"${appName}
   timeToLiveInSeconds = 900
+       encryption {
+        key = "Zovatyooe9d6vNI+dy1YMj7J3L9emXjQr6QtsL2a4LU="
```
Comment (drewmiley) [inline conf/application.conf:38]: Correct indentation

File: app/repositories/CacheMap.scala [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/*
```
Comment (drewmiley) [inline app/repositories/CacheMap.scala:1]: Is this needed?

File: app/repositories/DatedCacheMap.scala [hunk 4]
```diff
@@ -0,0 +1,62 @@
+/*
```
Comment (drewmiley) [inline app/repositories/DatedCacheMap.scala:1]: Is this needed?

File: app/models/ModelEncryption.scala [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+import play.api.libs.json.Json
+import repositories.DatedCacheMap
+import services.Encryption
+import uk.gov.hmrc.crypto.EncryptedValue
+
+import java.time.Instant
+
+object ModelEncryption {
+  def encryptDatedCacheMap(datedCacheMap: DatedCacheMap)(implicit encryption: Encryption): (String, Map[String, EncryptedValue], Instant) = {
```
Comment (drewmiley) [inline app/models/ModelEncryption.scala:27]: See other comments around this

**Review by drewmiley — COMMENTED** [review]
I think app/services/Encrpytion can be removed also, and encrypt/decrypt for both will have to be combined in ModelEncryption.scala

**Review by emanfalola9998 — APPROVED** [review]
LGTM

**Pull request merged**